### ISSUE TITLE
[Gemini Review]: notion-tables-td

### ISSUE THREAD

**Issue by NtFelix** [NtFelix/RMS#281]
![high](https://www.gstatic.com/codereviewagent/high-priority.svg)

This component renders all table cells as `<td>` elements. This is semantically incorrect for header cells and harms accessibility. If the table's `has_column_header` property is true, the cells in the first row (rendered in `<thead>`) should be `<th>` elements with a `scope="col"` attribute. If the table's `has_row_header` property is true, the first cell of each data row (in `<tbody>`) should be a `<th>` with a `scope="row"` attribute. The `TODO` on line 199 indicates this is a known issue.

_Originally posted by @gemini-code-assist[bot] in https://github.com/NtFelix/RMS/pull/278#discussion_r2191594751_
